.travis.yml

```diff
@@ -77,7 +28,57 @@ stages:
   - name: &build_unstable Deploy Pants Pex Unstable
     if: tag IS NOT present AND type NOT IN (pull_request, cron)
 
-default_test_config: &default_test_config
+# Travis cache config for jobs that build the native engine.
+native_engine_cache_config: &native_engine_cache_config
+  before_cache:
+    # Ensure permissions to do the below removals, which happen with or without caching enabled.
+    - sudo chown -R travis:travis "${HOME}" "${TRAVIS_BUILD_DIR}"
+    # Kill all python bytecode in our cached venvs.  Some files appear to
+    # get bytecode compiled in non-yet-understood circumstances leading to
+    # a full cache re-pack due to new bytecode files.
+    - find build-support -name "*.py[co]" -delete
+  cache:
+    # The default timeout is 180 seconds, and our larger cache uploads exceed this.
+    # TODO: Figure out why we have such large caches (2-7GB) and try to trim them.
+    timeout: 500
+    directories:
+      - ${HOME}/.cache/pants/rust/cargo
```

Comment:
We could shave 100M off this dir by replacing this with:
```
- ${HOME}/.cache/pants/rust/cargo/.crates.toml
- ${HOME}/.cache/pants/rust/cargo/git
- ${HOME}/.cache/pants/rust/cargo/registry
- ${HOME}/.cache/pants/rust/cargo/bin/cargo-ensure-installed
- ${HOME}/.cache/pants/rust/cargo/bin/cargo-ensure-prefix
```

at the expense of:
1. Hard coding some internals of cargo that may change
2. Hard-coding a list of things we `cargo install` (so that if we add more in the future, we'd need to update this list, or they'd be re-compiled)

I'm not sure what the cost of 100M here is, so can't really comment on the trade-off

Reply:
Makes sense, but again - let's experiment with this in a followup change whose only purpose is further cache tweaking? This was a tough change, and I'm leery of perturbing it any further when it's already a win as-is.